### DISCUSSION TITLE
Parsing through an entire dataset

### DISCUSSION THREAD

**Discussion by yhuwhai** [NVlabs/sionna#328] [Q&A] [answered]
Hi everyone,

So I used the ray tracing example in Sionna to generate CIRs dataset  for the neural SIMO receiver (also in Sionna). The CIRGenerator class outputs [a, Tau] without the batch size dimension. The CIRDataset class generates batches of [a,Tau] for sionna compliant channel model.  

I am wondering how the CIRDataset class parses the entire dataset. 

If the  CIRDataset class is called for a number of training iterations equal to dataset_size/batch_size, is it safe to say the entire dataset has been parsed sequentially?

Thank you in advance for your reply.

Best regards,

Uyoata

**Comment by faycalaa** [comment, marked as answer]
Hi,

`CIRDataset` loops through the entire dataset. Note that shuffling of the dataset is done for each epoch.

**Reply by yhuwhai** [reply]
Thanks @faycalaa for the response.  So for each epoch, CIRDataset goes through the batches of the dataset sequentially?

**Reply by faycalaa** [reply]
Yes
